File: spring-cloud-openfeign-demo/feign-consumer/src/main/java/com/livk/consumer/biz/ConsumerApp.java
package com.livk.consumer.biz;

import com.livk.common.LivkSpring;
import com.livk.provider.api.feign.UserRemoteService;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cache.annotation.EnableCaching;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
import org.springframework.cloud.openfeign.EnableFeignClients;

/**
 * <p>
 * ConsumerApp
 * </p>
 *
 * @author livk
 * @date 2021/12/6
 */
@EnableFeignClients(basePackageClasses = UserRemoteService.class)
@EnableCaching
@EnableDiscoveryClient
@SpringBootApplication
public class ConsumerApp {

	public static void main(String[] args) {
		LivkSpring.run(ConsumerApp.class, args);
	}

}

File: spring-cloud-openfeign/feign-consumer/src/main/kotlin/com/livk/consumer/biz/ConsumerApp.kt
package com.livk.consumer.biz

import com.livk.provider.api.feign.UserRemoteService
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cache.annotation.EnableCaching
import org.springframework.cloud.client.discovery.EnableDiscoveryClient
import org.springframework.cloud.openfeign.EnableFeignClients
import org.springframework.web.service.registry.ImportHttpServices

/**
 * @author livk
 */
@ImportHttpServices(basePackageClasses = [HttpUserRemoteService::class])
@EnableFeignClients(basePackageClasses = [UserRemoteService::class])
@EnableCaching
@EnableDiscoveryClient
@SpringBootApplication
open class ConsumerApp

fun main(args: Array<String>) {
    runApplication<ConsumerApp>(*args)
}
